adapter/src/test/java/com/twilio/kudu/sql/PaginationIT.java

```diff
@@ -437,4 +443,234 @@ public void testQueryMore() throws Exception {
     }
   }
 
+  @Test
+  public void testPaginationMultipleAccountsOrderedByTimestamp() throws Exception {
+    try (Connection conn = DriverManager.getConnection(JDBC_URL)) {
+      TimestampString lowerBoundDateInitiated = TimestampString.fromMillisSinceEpoch(T1);
+      TimestampString upperBoundDateInitiated = TimestampString.fromMillisSinceEpoch(T4);
+      String dateInitiatedOrder = descending ? "DESC" : "ASC";
+      String firstBatchSqlFormat = "SELECT * FROM %s WHERE (account_sid = '%s' OR account_sid = "
+          + "'%s') AND date_initiated >= TIMESTAMP'%s' AND date_initiated < TIMESTAMP'%s' "
+          + "ORDER BY date_initiated %s, transaction_id " + "LIMIT 7";
+      String firstBatchSql = String.format(firstBatchSqlFormat, tableName, ACCOUNT1, ACCOUNT2, lowerBoundDateInitiated,
+          upperBoundDateInitiated, dateInitiatedOrder);
+      ResultSet rs = conn.createStatement().executeQuery("EXPLAIN PLAN FOR " + firstBatchSql);
+      String plan = SqlUtil.getExplainPlan(rs);
+
+      String expectedPlanFormat = "KuduToEnumerableRel\n"
+          + "  KuduSortRel(sort0=[$1], sort1=[$2], dir0=[%s], dir1=[ASC], fetch=[7], " + "groupBySorted=[false])\n"
+          + "    KuduFilterRel(ScanToken 1=[account_sid EQUAL %s, date_initiated GREATER_EQUAL %d, "
```

Comment:
qq, since kudusortrel is not pushed to each of the scanners for accountsids, we would end up having to read all data for the accountsids for the dates? Or the optimization there would depend on how we pass dates in the `nextPageOffset`?

Reply:
When you pass in a row value constructor that context the next page the scan token will add the rvc as a filter (for eg in the plan for the next page query below).